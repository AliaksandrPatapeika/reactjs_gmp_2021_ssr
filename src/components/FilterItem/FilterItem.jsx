import './FilterItem.less';

import PropTypes from 'prop-types';
import React, {useMemo} from 'react';

const FilterItem = ({label, isActive, setFilter}) => {
  const className = useMemo(() => (isActive ? 'filterItem active' : 'filterItem'), [isActive]);

  return (
    <li role="menuitem" tabIndex="0" className={className} onClick={setFilter} onKeyPress={setFilter}>
      {label}
    </li>
  );
};

FilterItem.propTypes = {
  label: PropTypes.string.isRequired,
  isActive: PropTypes.bool.isRequired,
  setFilter: PropTypes.func.isRequired
};

export default FilterItem;
