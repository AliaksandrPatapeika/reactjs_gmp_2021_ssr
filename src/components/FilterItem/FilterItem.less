@import "../../less/variables";

.filterItem {
	cursor: pointer;
	padding: 0 12px 17px 0;
	color: @white;
	font-size: 20px;
	font-family: @NetflixSans-Regular;

	&:first-of-type {
		padding-left: 0;
	}

	&:hover,
	&:focus {
		color: @red;
	}
}

.active {
	border-bottom: 3px solid @red;
}
