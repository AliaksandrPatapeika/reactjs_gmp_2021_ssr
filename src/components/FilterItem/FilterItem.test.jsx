import {noop} from 'lodash';
import React from 'react';
import renderer from 'react-test-renderer';

import FilterItem from './FilterItem';

describe('FilterItem', () => {
  test('renders correctly', () => {
    const filterItem = renderer
      .create(<FilterItem label="ALL" isActive setFilter={noop} />)
      .toJSON();

    expect(filterItem).toMatchSnapshot();
  });
});
