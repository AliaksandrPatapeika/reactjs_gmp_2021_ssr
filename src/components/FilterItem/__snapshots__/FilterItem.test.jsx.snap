// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterItem renders correctly 1`] = `
<li
  className="filterItem active"
  onClick={[Function]}
  onKeyPress={[Function]}
  role="menuitem"
  tabIndex="0"
>
  ALL
</li>
`;
